### PR TITLE
tests: Exit riscv_asmtest script with simulator status code

### PR DESCRIPTION
**Pull request by rogerchang23424**
It will be helpful to check if the instruction simulate well

Change-Id: I5faa435fad79601682126ee7978d8444093df900

### CODE REVIEW

**Comment by rogerchang23424** [comment]
Blocked by #890

**Review by powerjg — COMMENTED** [review]
lgtm after a rebase

**Review by powerjg — APPROVED** [review]

**Pull request merged**
